src/stack.py

```diff
@@ -0,0 +1,36 @@
+"""Module with implementation of stack data structure."""
+from linked_list import LinkedList
+
+
+class Stack(object):
+    """Stack data structure class."""
+
+    def __init__(self, iterable=None):
+        """Stack constructor."""
+        self._linkedlist = LinkedList(iterable)
+        self._update_attr()
+
+    def push(self, val):
+        """Add value to top of stack."""
+        self._linkedlist.push(val)
+        self._update_attr()
+
+    def pop(self):
+        """Remove and return top of stack."""
+        try:
+            res = self._linkedlist.pop()
+            self._update_attr()
+            return res
+        except IndexError:
+            raise IndexError("Cannot pop from empty stack.")
+
+    def size(self):
+        """Return size of stack."""
+        return self._linkedlist.size()
+
+    def __len__(self):
+        """Return size of stack with len builtin."""
+        return self.size()
+
+    def _update_attr(self):
```

Comment:
What is the purpose of this method? Is it actually helping anything? What is the `top` attribute, and why would you need it?